api/v1alpha1/volumereplication_types.go

```diff
@@ -20,20 +20,77 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// ImageState represents the replication operations to be performed on the image
+type ImageState string
+
+const (
+	// Primary ImageState enables mirroring and promotes the image to primary
+	Primary ImageState = "primary"
+
+	// Secondary ImageState demotes the image to secondary and resyncs the image if out of sync
+	Secondary ImageState = "secondary"
+
+	// Resync option resyncs the image
+	Resync ImageState = "resync"
+)
+
+// ReplicationState captures the latest state of the replication operation
+type ReplicationState string
+
+const (
+	// MirroringEnabled means mirroring is enabled between two sites
+	MirroringEnabled ReplicationState = "MirroringEnabled"
+
+	// ImagePromoted means image was promoted to primary
+	ImagePromoted ReplicationState = "Promoted"
+
+	// ImageDemoted means image was demoted to secondary
+	ImageDemoted ReplicationState = "Demoted"
+
+	// ImageResynced means image was resynced
+	ImageResynced ReplicationState = "Resynced"
+
+	// MirroringFailed means mirroring operation was failed
+	MirroringFailed ReplicationState = "MirroringFailed"
+
+	// ImagePromotionFailed means image promotion failed
+	ImagePromotionFailed ReplicationState = "PromotionFailed"
+
+	// ImageDemotionFailed means image demotion failed
+	ImageDemotionFailed ReplicationState = "DemotionFailed"
+
+	// ImageResyncFailed means image resync operation failed
+	ImageResyncFailed ReplicationState = "ResyncFailed"
+)
+
+// DataSourceSpec is
+type DataSourceSpec struct {
+	APIGroup string `json:"apiGroup"`
+	Kind     string `json:"kind"`
+	// Name is the PVC name associated with the image
+	Name string `json:"name"`
+}
 
 // VolumeReplicationSpec defines the desired state of VolumeReplication
 type VolumeReplicationSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// VolumeReplicationClass is the VolumeReplicationClass name for this VolumeReplication resource
+	VolumeReplicationClass string `json:"volumeReplicationClass"`
+
+	// ImageState represents the replication operation to be performed on the image.
+	// Supported operations are "primary", "secondary" and "resync"
+	ImageState ImageState `json:"imageState"`
 
+	// DataSource represents the PVC name associated with the image
+	DataSource DataSourceSpec `json:"dataSource"`
```

Comment:
(asking for clarification) is there a reason not to use DataSource *corev1.TypedLocalObjectReference?